plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -382,10 +379,16 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         if (TIME_WITH_TIME_ZONE.equals(type) || TIMESTAMP_TZ_MILLIS.equals(type)) {
             throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
         }
-        if (TIMESTAMP_MILLIS.equals(type)) {
-            // TODO use `timestampWriteFunction` (https://github.com/trinodb/trino/issues/6910)
-            return WriteMapping.longMapping("datetime(3)", timestampWriteFunction(TIMESTAMP_MILLIS));
+
+        if (type instanceof TimestampType) {
+            TimestampType timestampType = (TimestampType) type;
+            if (timestampType.getPrecision() <= MAX_SUPPORTED_TIMESTAMP_PRECISION) {
+                verify(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION, "Unexpected timestamp precision %s calculated for supported datetime column precision %s", timestampType.getPrecision(), MAX_SUPPORTED_TIMESTAMP_PRECISION);
```

Comment:
i am not sure i understand the message here.

since it's actually a "compile time sanity check" (basically ensuring that MAX_SUPPORTED_TIMESTAMP_PRECISION = MAX_SHORT_PRECISION), we can afford going without a message here

```
verify(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION);
```